lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb

```diff
@@ -1543,7 +1543,10 @@ fc_extras
             bounds_data = cf_bounds_var[:]
             # Gracefully fill bounds masked array.
             if ma.isMaskedArray(bounds_data):
-                bounds_data = ma.filled(bounds_data)
+                if not ma.isMaskedArray(points_data):
```

Comment:
By this point in the function `points_data` has been filled, so `ma.isMaskedArray(points_data)` will always return `False`. You need to record the value of `ma.isMaskedArray(points_data)` before the array is filled.